### PR TITLE
[8.0] Add timesheet project access restriction module

### PR DESCRIPTION
**Pull request by adrienpeiffer**
This module adds some record rule to avoid to access on not authorized project.
- [ ] Depends on https://github.com/OCA/account-analytic/pull/44

### CODE REVIEW

File: hr_timesheet_project_access_restriction/README.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License AGPL-3
+
+===================================
+HR Timesheet No Closed Project-Task
```
Comment (lmignon) [inline hr_timesheet_project_access_restriction/README.rst]: The title comes from https://github.com/acsone/acsone-addons/pull/76 :smirk:

File: hr_timesheet_project_access_restriction/README.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+===================================
+
+This module adds some record rule to avoid to access on not authorized project.
```
Comment (lmignon) [inline hr_timesheet_project_access_restriction/README.rst]: The module description should be improved to describe more precisely the way the security can be managed on the project.... I'm not an expert of this kind of description but something like

```
This module allow you to manage the access to projects by project:
An employee defined as 'manager' on a project can:
*  Manage the project,
*  See analytic accounts related to the project,
*  Read all timesheet lines for the project .

An employee member of a project can:
* See the project,
* See analytic accounts related to the project and therefore create timesheet lines on these analytic accounts,
* See their own timesheet line related to the project.

```

File: hr_timesheet_project_access_restriction/security/hr_timesheet_project_access_restriction_security.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                            ('members', 'in', [user.id])
+                                        ]</field>
+        </record>
+        <!-- Overrride standard record rule to allow simple employee to access
+        on analytic account in which they are member on the related project-->
+        <record id="analytic_project_public_members_rule" model="ir.rule">
+            <field name="name">Analytic Account - Project: employees: public, employees, followers, members or project responsible</field>
+            <field name="domain_force">['|', '|', '|',
+                                            ('project_ids.privacy_visibility', 'in', ['public', 'employees']),
+                                            ('project_ids.message_follower_ids', 'in', [user.partner_id.id]),
```
Comment (lmignon) [inline hr_timesheet_project_access_restriction/security/hr_timesheet_project_access_restriction_security.xml]: What's the difference between followers and members? Do we have to grant access to the followers on the analytic accounts?

**Comment by adrienpeiffer** [comment]
Thanks @lmignon

**Comment by lmignon** [comment]
:+1:  LGTM

**Pull request merged**